backend/routes/auth.js

```diff
@@ -0,0 +1,147 @@
+const express = require("express");
+const { check, validationResult } = require("express-validator/check");
+const bcrypt = require("bcryptjs");
+const jwt = require("jsonwebtoken");
+const router = express.Router();
+
+const { User } = require("../models/user");
+
+/**
+ * @method - POST
+ * @param - /signup
+ * @description - User SignUp
+ */
+
+router.post(
+  "/signup",
+  [
+    check("username", "Please Enter a Valid Username").not().isEmpty(),
+    // check("email", "Please enter a valid email").isEmail(),
+    check("password", "Please enter a valid password").isLength({
+      min: 1,
+    }),
+  ],
+  async (req, res) => {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      return res.status(400).json({
+        errors: errors.array(),
+      });
+    }
+
+    const { username, password } = req.body;
+    try {
+      let user = await User.findOne({
+        username,
+      });
+      if (user) {
+        return res.status(400).json({
+          message: "User Already Exists",
```

Comment:
Can we change "User" to "Username", since they are picking a username, we tell them that the username is already being used.